packages/launcher/lib/detect.ts

```diff
@@ -25,20 +26,24 @@ type HasVersion = Omit<Partial<FoundBrowser>, 'version' | 'name'> & {
   name: string
 }
 
-export const setMajorVersion = <T extends HasVersion>(browser: T): T => {
-  const ver = browser.version.split('.')[0] ?? browser.version
-  const majorVersion = parseInt(ver) || browser.version
-
-  const unsupportedVersion = browser.minSupportedVersion && majorVersion < browser.minSupportedVersion
+const getMajorVersion = (version: string): string => {
+  return version.split('.')[0] ?? version
+}
 
-  const foundBrowser = extend({}, browser, { majorVersion })
+// Determines if found browser is supported by Cypress. If found to be
+// unsupported, the the browser will be unavailable for selection and
+// will present the determined warning message to the user.
+const validateCypressSupport = (validator: BrowserValidator | undefined, browser: FoundBrowser, platform: NodeJS.Platform) => {
+  // If no validator parameter is provided, we fall back to validating against
+  // the browser's minimum supported version
+  const { isSupported, warningMessage } = (validator || validateMinVersion)(browser, platform)
 
-  if (unsupportedVersion) {
-    foundBrowser.unsupportedVersion = true
-    foundBrowser.warning = `Cypress does not support running ${browser.displayName} version ${majorVersion}. To use ${browser.displayName} with Cypress, install a version of ${browser.displayName} newer than or equal to ${browser.minSupportedVersion}.`
+  if (isSupported) {
+    return
   }
 
-  return foundBrowser
+  browser.unsupportedVersion = true
+  browser.warning = warningMessage
```

Comment:
I'm noticing now that I abort early if isSupported is truthy; anyone see value 'it is supported but still has a warning' use case?